http-clients/url-connection-client/src/main/java/software/amazon/awssdk/http/urlconnection/UrlConnectionHttpClient.java

```diff
@@ -199,6 +210,31 @@ public void setConnectionTimeout(Duration connectionTimeout) {
             connectionTimeout(connectionTimeout);
         }
 
+        /**
+         * A function used to create a {@link HttpURLConnection} from a {@link URL}.
+         *
+         * By default {@link URL#openConnection()} is used.
+         *
+         * @param connectionCreator the function to call to create the connection
+         * @return this object for method chaining
+         */
+        @Override
+        public Builder connectionCreator(Function<URL, HttpURLConnection> connectionCreator) {
+            connectionCreator = connectionCreator;
```

Comment:
SpotBugs flagged this dead store. Can we fix and add a test case?

Reply:
Once we agree this is the right approach I'm happy to add a test.

Reply:
fixed the actual spot-bugs issue - what a noob mistake - been away from Java too long :P